astropy/io/fits/tests/test_fitstime.py

```diff
@@ -0,0 +1,158 @@
+
+
+class TestFitsTime(FitsTestCase):
+
+    def setup_class(self):
+        self.time = ['1999-01-01T00:00:00.123456789', '2010-01-01T00:00:00']
+
+    def test_is_time_column_keyword(self):
+        # Time column keyword without column number
+        assert is_time_column_keyword('TRPOS') is False
+
+        # Global time column keyword
+        assert is_time_column_keyword('TIMESYS') is False
+
+        # Valid time column keyword
+        assert is_time_column_keyword('TRPOS12') is True
+
+    @pytest.mark.parametrize('table_types', (Table, QTable))
+    def test_time_to_fits_loc(self, table_types):
+        """
+        Test all the unusual conditions for locations of `Time` columns in a `Table`.
+        """
+        t = table_types()
+        t['a'] = Time(self.time, format='isot', scale='utc')
+        t['b'] = Time(self.time, format='isot', scale='tt')
+
+        # Check that vectorized location raises an exception
+        t['a'].location = EarthLocation([1,2], [2,3], [3,4,])
+
+        with pytest.raises(ValueError) as err:
+            table, hdr = time_to_fits(t)
+        assert 'Vectorized Location of Time Column' in str(err.value)
+
+        # Check that multiple Time columns with different locations raise an exception
+        t['a'].location = EarthLocation(1, 2, 3)
+        t['b'].location = EarthLocation(2, 3, 4)
+
+        with pytest.raises(ValueError) as err:
+            table, hdr = time_to_fits(t)
+            assert 'Multiple Time Columns with different geocentric' in str(err.value)
+
+        # Check that Time column with no location specified will assume global location
+        t['b'].location = None
+
+        with catch_warnings() as w:
+            table, hdr = time_to_fits(t)
+            assert len(w) == 1
+            assert str(w[0].message).startswith('Time Column "b" has no specified location,'
+                                                ' but global Time Position is present')
+
+        # Check that multiple Time columns with same location can be written
+        t['b'].location = EarthLocation(1, 2, 3)
+
+        with catch_warnings() as w:
+            table, hdr = time_to_fits(t)
+            assert len(w) == 0
+
+        # Check compatibility of Time Scales and Reference Positions
+
+        for scale in BARYCENTRIC_SCALES:
+            t.replace_column('a', getattr(t['a'], scale))
+            with catch_warnings() as w:
+                table, hdr = time_to_fits(t)
+                assert len(w) == 1
+                assert str(w[0].message).startswith('Earth Location "TOPOCENTER" for Time Column')
+
+    @pytest.mark.parametrize('table_types', (Table, QTable))
+    def test_time_to_fits_header(self, table_types):
+        """
+        Test the header returned by `time_to_fits` explicitly.
+        """
+        t = table_types()
+        t['a'] = Time(self.time, format='isot', scale='utc',
+                      location=EarthLocation(-2446353.80003635,
```

Comment:
Make these into round integers so testing is less fragile (i.e. you get exact floating point representations).